pkg/utils/k8sutils.go

```diff
@@ -141,54 +137,19 @@ func GetMD5Hash(text string) string {
 	return hex.EncodeToString(hash[:])
 }
 
-func CreateOrReplace(ctx context.Context, c client.Client, obj client.Object, f controllerutil.MutateFn) error {
-	key := client.ObjectKeyFromObject(obj)
-	if err := c.Get(ctx, key, obj); err != nil {
-		if !errors.IsNotFound(err) {
-			return err
-		}
-		if err := mutate(f, key, obj); err != nil {
-			return err
-		}
-		if err := c.Create(ctx, obj); err != nil {
-			return err
-		}
-		return nil
-	}
-
-	existing := obj.DeepCopyObject()
-	if err := mutate(f, key, obj); err != nil {
-		return err
-	}
-
-	if equality.Semantic.DeepEqual(existing, obj) {
-		return nil
-	}
-
-	if err := c.Delete(ctx, obj); err != nil {
-		return err
-	}
-
-	// k8s doesn't allow us to create objects when resourceVersion is set, as we are DeepCopying the
-	// object, the resource version also gets copied, hence we need to set it to empty before creating it
-	obj.SetResourceVersion("")
-	if err := c.Create(ctx, obj); err != nil {
-		return err
-	}
-	return nil
+func GetClusterResourceQuotaName(name string) string {
+	return fmt.Sprintf("storage-client-%s-resourceqouta", GetMD5Hash(name))
 }
 
-// mutate wraps a MutateFn and applies validation to its result.
-func mutate(f controllerutil.MutateFn, key client.ObjectKey, obj client.Object) error {
-	if err := f(); err != nil {
-		return err
+func IsForbiddenError(err error) bool {
+	if err == nil {
+		return false
 	}
-	if newKey := client.ObjectKeyFromObject(obj); key != newKey {
-		return fmt.Errorf("MutateFn cannot mutate object name and/or object namespace")
+	statusErr := err.(*errors.StatusError)
+	for i := range statusErr.ErrStatus.Details.Causes {
+		if statusErr.ErrStatus.Details.Causes[i].Type == metav1.CauseTypeForbidden {
+			return true
+		}
 	}
-	return nil
-}
-
-func GetClusterResourceQuotaName(name string) string {
-	return fmt.Sprintf("storage-client-%s-resourceqouta", GetMD5Hash(name))
+	return false
```

Comment:
- For a function returning a boolean, please prefer only one return of true value and only one return of false value.
- Please use a non panicing down cast 
```suggestion
	statusErr, ok := err.(*errors.StatusError)
	if ok {
		for i := range statusErr.ErrStatus.Details.Causes {
			if statusErr.ErrStatus.Details.Causes[i].Type == metav1.CauseTypeForbidden {
				return true
			}
		}
	}
	return false
```